### ISSUE TITLE
Support "mp.get_script_file()" in Lua scripting

### ISSUE THREAD

**Issue by snowman** [mpv-player/mpv#8614]
### Expected behavior of the wanted feature
Currently, only JavaScript support `mp.get_script_file()` in `Additional utilities` section of mpv manpage.

### Workaround
```lua
local current_script_file = debug.getinfo(1).source
```

**Comment by avih** [comment]
I'm not sure if your workaround is the same functionality as in js - in js it's always the main script file, even in a file which gets `require`'ed (and it's used mainly at the implementation of `require`).

Do you have a use case where the JS functionality would be useful in lua?

**Comment by snowman** [comment]
If you ask me the use case, that's a long story.

I have used `loader.lua` to load other user scripts.

```lua
$ cd MPV_CONFIG_DIR/scripts
$ git clone https://github.com/occivink/mpv-gallery-view vendor.disable/occivink/mpv-gallery-view
$ cat loader.lua
function load_vendor_script(script)
   mp.commandv("load-script", "~~/scripts/vendor.disable/" .. script)
end

load_vendor_script("occivink/mpv-gallery-view/scripts/contact-sheet.lua")
load_vendor_script("occivink/mpv-gallery-view/scripts/gallery-thumbgen.lua")
load_vendor_script("occivink/mpv-gallery-view/scripts/playlist-view.lua")
$ tree
.
├── loader.lua
└── vendor.disable
    └── occivink
        └── mpv-gallery-view
            ├── LICENSE
            ├── README.md
            ├── script-opts
            │   ├── contact_sheet.conf
            │   ├── gallery_worker.conf
            │   └── playlist_view.conf
            └── scripts
                ├── contact-sheet.lua
                ├── gallery-thumbgen.lua
                ├── lib.disable
                │   └── gallery.lua
                └── playlist-view.lua
```

File `contact-sheet.lua`:

```lua
-- contact-sheet.lua using un-documented API: "mp.find_config_file" 
local lib = mp.find_config_file('scripts/lib.disable')
if not lib then
    return
end
...
```

Currently, I use
`local lib = debug.getinfo(1).source:match("@?(.*/)") .. 'lib.disable'`
to get the `lib.disable` relative to the current running script instead of fixed location `MPV_CONFIG_DIR`

And as you say, the workaround is not the same as `mp.get_script_file()` in javascript,
and even though you add the workaround solution for Lua,
I have to send PR to "occivink/mpv-gallery-view" which may not be merged in.

So, that's it.

Thank you, I'll close this issue.

**Comment by avih** [comment]
Sorry, I just don't understand. I don't object to adding it, and it's probably very easy to do, but I prefer to have a good reason for it first, and so far I don't understand your explanation.

**Comment by avih** [comment]
If you still have your issue, you might be able to solve it by loading your script as a directory-script (a directory in `scripts/` which contains `main.lua`).

If you do that, then the directory path is available via `mp.get_script_directory()`. This works in both lua and js.
